File: server/src/main/java/com/preponderous/roam/model/entity/Wood.java
package com.preponderous.roam.model.entity;

import com.preponderous.roam.model.Entity;

/**
 * Represents a wood resource entity.
 * Wood can be gathered from trees and used for crafting.
 * 
 * @author Daniel McCoy Stephenson
 */
public class Wood extends Entity {
    private int quantity;

    public Wood() {
        super("Wood", "assets/images/wood.png");
        this.quantity = 1;
        this.setSolid(false);
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}

File: src/entity/wood.py
from entity.drawableEntity import DrawableEntity


# @author Daniel McCoy Stephenson
# Generic wood entity for server-backed architecture
class Wood(DrawableEntity):
    def __init__(self):
        # Use oak wood image as generic wood
        DrawableEntity.__init__(self, "Wood", "assets/images/oakWood.png")
        self.solid = True

    def isSolid(self):
        return self.solid
